### DISCUSSION TITLE
In Blazor, what is an "interactive render mode" and how do I use it?

### DISCUSSION THREAD

**Discussion by MisinformedDNA** [dotnet/aspnetcore#50173] [Q&A]
I keep reading "interactive render mode" and I'm not sure what to think. Most web pages are interactive, so we would almost always want to enable this, right. Non-interactive render mode seems like it would be an edge case?

Also, how do I enable "interactive render mode" for WASM? In @mkArtak's [example](https://github.com/mkArtakMSFT/BlazorWasmClientInteractivity/tree/main), he creates two projects, one for WASM and one for server, but I thought the whole point of Blazor "United" was that I only needed one project to do either.

**Comment by KennethHoff** [comment]
A large number of websites are non-interactive. I don't know if it's a majority or not, but I'm willing to bet it actually is. A majority of websites are definitely "might as well be non-interactive" if not literally. 

WASM, by definition, will always be interactive.

All of this is going from the assumption that "interactive" refers to all non-HTML interactions(basically, JavaScript/WASM) and not things like forms and links.

**Reply by Xyncgas** [reply]
what is interactive? Is single page web app interactive? If I am not making request to server is the UI interactive? So reactive UI using javascript is interactive, may be if I am making a little request but not too much for example signalR that's still interactive

**Reply by KennethHoff** [reply]
Anything that mutates the DOM outside of the initial Document page load is interaction. Just to reiterate my first point: An overwhelming majority of _modern_ websites are interactive, but old websites are generally non-interactive.

My second point ("A majority might as well be") was about the fact that most pages uses Vue, React, etc.. even though they're 0 reason to do so.
